esp-hal/src/spi/master.rs

```diff
@@ -416,13 +416,32 @@ impl Address {
     }
 }
 
+/// SPI clock source.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum ClockSource {
+    /// Use the APB clock.
+    Apb,
+    // #[cfg(any(esp32c2, esp32c3, esp32s3))]
+    // Xtal,
+}
+
 /// SPI peripheral configuration
 #[derive(Clone, Copy, Debug, PartialEq, Eq, procmacros::BuilderLite)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 #[non_exhaustive]
 pub struct Config {
-    /// SPI bus clock frequency.
-    pub frequency: HertzU32,
```

Comment:
If this is now private, we probably want to add a `frequency()` getter to access the value?

Reply:
Yes, the question is whether we want to generate the accessors. I'd prefer hiding fields and generating everything, as that gives us the most freedom in the future w.r.t. additional logic on the fields, but I left this intentionally open for debate.

Reply:
From my perspective, consistency is key. I think either all fields should be public, _or_ all fields should be private with a getter.

Given that we now have some private fields that _can't_ be public else we may invalidate some internal book keeping, I'd say we should switch to the latter. We can do this here, or file an issue to do it in a follow up.

Reply:
Will be done here, this needs to be a complete package IMO :)